### PR TITLE
add shouldfocus related to #253

### PR DESCRIPTION
**Pull request by Freundschaft**
Add option to set [shouldFocus](https://developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindowOpenOptions.shouldFocus)

### CODE REVIEW

**Comment by usefulthink** [comment]
Awesome, thanks! I think the `shouldFocus` prop needs to be added to the `InfoWindowProps` type, since that only includes the `google.maps.InfoWindowOptions`, not the `google.maps.InfoWindowOpenOptions`.

Also please check the formatting with prettier (`npm run test:prettier` to check and `npx prettier --write ./src/components/info-window.tsx` to fix).

**Comment by Freundschaft** [comment]
whoops youre right ill fix and resubmit

**Comment by usefulthink** [comment]
> Could it also be considered to add InfoWindowOpenOptions as a prop entirely in case the interface changes in the future?

I don't think that makes a lot of sense, since the open-options have to always be considered separately as you did here. So new options would just cause unexpected behaviour until they are implemented.

File: src/components/info-window.tsx [hunk 1]
```diff
@@ -16,13 +16,15 @@ import {GoogleMapsContext} from './map';
 export type InfoWindowProps = google.maps.InfoWindowOptions & {
   onCloseClick?: () => void;
   anchor?: google.maps.Marker | google.maps.marker.AdvancedMarkerElement | null;
+  shouldFocus?: boolean | null;
```
Comment (usefulthink) [inline src/components/info-window.tsx:19]: I think we can leave away the `| null` here. Optional and boolean should be enough.

```suggestion
  shouldFocus?: boolean;
```

**Review by usefulthink — COMMENTED** [review]
except for the type it's perfect.

**Pull request merged**